src/analysisd/decoders/dbsync.c

```diff
@@ -123,6 +130,13 @@ static void dispatch_check(dbsync_context_t * ctx, const char * command) {
     os_malloc(OS_MAXSTR, query);
     os_malloc(OS_MAXSTR, response);
 
+    ComponentType component_type = validate_component(ctx->component);
```

Comment:
We can remove this and centralize the component check at `DispatchDBSync()`

Reply:
Moved!